clang/test/dpct/helper_files_ref/include/dpl_extras/functional.h

```diff
@@ -22,6 +22,8 @@
 
 namespace dpct {
 
+struct null_type {};
```

Comment:
this PR doesn't show the use of null_type{}.  So not sure how it is used and why it is necessary to add null_type.

Reply:
I've just added a new PR into SYCLomatic-test which shows the intended usage of `dpct::null_type`.  

https://github.com/oneapi-src/SYCLomatic-test/pull/219

It will also likely be used in future dpct helper header APIs to a similar effect.